crates/router/src/services/authentication.rs

```diff
@@ -352,7 +359,12 @@ where
             .map_err(|_| errors::ApiErrorResponse::MissingRequiredField {
                 field_name: "client_secret",
             })?;
-        return Ok((Box::new(PublishableKeyAuth), api::AuthFlow::Client));
+        return Ok((
+            Box::new(PublishableKeyAuth {
+                client_secret: payload.get_client_secret().cloned(),
```

Comment:
Are we reading this value anywhere?

Reply:
This is something that was already there haven't made any changes in this and it is used in server wrap functions moreover It was already there

Reply:
This change is being added in this commit. a field called `client_secret` is added to `PublishableKeyAuth` in this PR.